### PR TITLE
Movement voyager integration

### PR DESCRIPTION
**Pull request by aeryz**

### CODE REVIEW

File: lib/cometbft-rpc/src/lib.rs [hunk 1]
```diff
@@ -329,7 +329,7 @@ impl ClientT for ClientInner {
 //             )
 //             .await;
 
-//         dbg!(result);
+//         bg!(result);
```
Comment (aeryz) [inline lib/cometbft-rpc/src/lib.rs:332]: wazzaaap

File: cosmwasm/ucs01-relay/src/protocol.rs [hunk 2]
```diff
@@ -1127,6 +1127,7 @@ impl<'a> TransferProtocol for Ucs01Protocol<'a> {
 }
 
 #[cfg(test)]
+#[allow(deprecated)] // TODO: Remove usage of mock_info
```
Comment (benluelo) [inline cosmwasm/ucs01-relay/src/protocol.rs:1130]: remove comment

File: lib/unionlabs/src/aptos/block_info.rs [hunk 3]
```diff
@@ -32,9 +35,9 @@ pub struct BlockInfo {
     /// The consensus protocol is executed in rounds, which monotonically increase per epoch.
     pub round: Round,
     /// The identifier (hash) of the block.
-    pub id: H256,
+    pub id: Hash<32, HexUnprefixed>,
     /// The accumulator root hash after executing this block.
-    pub executed_state_id: H256,
+    pub executed_state_id: Hash<32, HexUnprefixed>,
```
Comment (benluelo) [inline lib/unionlabs/src/aptos/block_info.rs:40]: NOTE: replace these with the proper H256 type alias when rebasing #2984 once this is merged

File: lib/unionlabs/src/aptos/sparse_merkle_proof.rs [hunk 4]
```diff
@@ -24,13 +27,13 @@ pub struct SparseMerkleProof {
...
-    pub siblings: Vec<H256>,
+    pub siblings: Vec<Hash<32, HexUnprefixed>>,
 }
 
 #[derive(Clone, Copy, Debug, Eq, PartialEq, Serialize, Deserialize)]
 pub struct SparseMerkleLeafNode {
-    pub key: H256,
-    pub value_hash: H256,
+    pub key: Hash<32, HexUnprefixed>,
+    pub value_hash: Hash<32, HexUnprefixed>,
```
Comment (benluelo) [inline lib/unionlabs/src/aptos/sparse_merkle_proof.rs:36]: NOTE: same comment here for #2984

File: lib/unionlabs/src/errors.rs [hunk 5]
```diff
@@ -20,6 +20,11 @@ macro_rules! required {
                 $field
             ))))
     };
+    ($field:ident) => {
+        $field.ok_or(<Self::Error>::MissingField(MissingField(stringify!(
+            $field
+        ))))
+    };
```
Comment (benluelo) [inline lib/unionlabs/src/errors.rs:27]: 👍

File: lib/voyager-message/src/plugin.rs
Comment (benluelo) [inline lib/voyager-message/src/plugin.rs:1]: bad rebase, remove

File: move/move-ibc/ibc-module.json
Comment (benluelo) [inline move/move-ibc/ibc-module.json:1]: yeet this

File: move/move-ibc/sources/bcs_utils.move [hunk 6]
```diff
@@ -77,6 +81,12 @@ module IBC::bcs_utils {
         from_bcs::to_bytes(vector::slice(&buf.inner, buf.cursor - (length as u64) - n_read, buf.cursor))
     }
 
+    // /// Peel a vector<u8>
```
Comment (benluelo) [inline move/move-ibc/sources/bcs_utils.move:84]: r u sure about that

File: move/move-ibc/sources/bcs_utils.move [hunk 7]
```diff
@@ -53,7 +57,7 @@ module IBC::bcs_utils {
 
     /// Peel a u32
     public fun peel_u32(buf: &mut BcsBuf): u32 {
-        buf.cursor = buf.cursor + 8;
+        buf.cursor = buf.cursor + 4;
```
Comment (benluelo) [inline move/move-ibc/sources/bcs_utils.move:60]: i still can't believe how much of an unbelievable rug this was

File: voyager/devnet-config.json
Comment (benluelo) [inline voyager/devnet-config.json:1]: let's just use `./target` lol

File: voyager/modules/client/cometbls/src/main.rs [hunk 8]
```diff
@@ -308,10 +310,20 @@ impl ClientModuleServer<ModuleData, ModuleCall, ModuleCallback> for ModuleServer
...
-                SupportedIbcInterface::IbcMoveAptos => header.encode_as::<Bcs>(),
-            })
+            .map(|mut header| match self.ctx.ibc_interface {
+                SupportedIbcInterface::IbcSolidity => Ok(header.encode_as::<EthAbi>()),
+                SupportedIbcInterface::IbcMoveAptos => {
+                    header.zero_knowledge_proof =
+                        reencode_zkp_for_move(&header.zero_knowledge_proof).map_err(|e| {
+                            ErrorObject::owned(
+                                FATAL_JSONRPC_ERROR_CODE,
+                                format!("unable to decode zkp: {}", e),
```
Comment (benluelo) [inline voyager/modules/client/cometbls/src/main.rs:320]: decode? or reencode?

Comment (aeryz) [reply voyager/modules/client/cometbls/src/main.rs:320]: its reencode bc we decode then reencode it using ark-bn254

**Review by benluelo — APPROVED** [review]

**Pull request merged**